### PR TITLE
Recommend only venv-based dev environment (drop conda)

### LINKED ISSUE #7478: Setting up conda environment for development of skimage throws a package not found error
### Description:

Following the conda environment setup instructions at: https://scikit-image.org/docs/stable/user_guide/install.html#id4 fails at the line ` conda install -c conda-forge --file requirements/default.txt` with a package not found error:

```
PackagesNotFoundError: The following packages are not available from current channels:

  - lazy-loader[version='>=0.4']

Current channels:

  - https://conda.anaconda.org/conda-forge
  - defaults
```

The lazy loader package is available on conda-forge ([here](https://anaconda.org/conda-forge/lazy_loader)) but its installation is listed as `conda install lazy_loader` instead of `conda install lazy-loader`. 

### Way to reproduce:

Follow the conda environment setup instructions for skimage dev at https://scikit-image.org/docs/stable/user_guide/install.html#id4. 

### Version information:

```Shell
conda version: 24.5.0
```

### PR DESCRIPTION
**Pull request by lagru**
## Description

Closes #7478. 

I propose to drop conda / conda-forge. We only really test with dependencies from PyPI in our CI. Our dev workflow heavily depends on spin. There are command line options in several commands that use pip and PyPI to handle dependencies. It also addresses issues like in #7478 where we run into the fact that package names (and syntax) are different between pip and conda.

For other reasons to keep to the pip-venv route, I can recommend reading [Back to basics with pip and venv](https://www.bitecode.dev/p/back-to-basics-with-pip-and-venv). While that article has some strong opinions, I tend to agree with the general direction. :D

I also took the liberty to add a few tweaks here and there. Happy to revert if they are controversial.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpydoc/blob/main/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
...
```

### CODE REVIEW

**Review by jarrodmillman — APPROVED** [review]

**Comment by hmaarrfk** [comment]
This is a mistake in my mind

conda-forge provides a lot of high quality scientific packages for video processing. 

I’ve personally worked to package ffmpeg and hdf5 over the years which are both extremely popular for video and image storage.

we also worked to provide some of the first packages for arm processors that were widely distributed. 

I think that the issues you listed are all resolvable and I would be happy to make PRs to resolve them. 

Ultimately, if we want to start to ship more code that depends on just in time compilation, llvm and other c like packages from conda forge are going to become more important.

File: INSTALL.rst
Comment (hmaarrfk) [inline INSTALL.rst:182]: This line in particular is much easier with conda-forge (in my opinion) when trying to ensure a consistent set of compilers and feature are used across different versions of linux.

File: INSTALL.rst
Comment (hmaarrfk) [inline INSTALL.rst:279]: I understand that these two files are difficult to keep in sync. with an additional conda environment file it doesn't get easier.... i know...

But scipy has this file,
https://github.com/scipy/scipy/blob/main/environment.yml#L8

and it might be appropriate to just keep it a `default + testing` dependencies for simplicity

**Comment by lagru** [comment]
@hmaarrfk I'm totally onboard with your reasoning that supporting conda / conda-forge for end-users is worthwile; we want to reach as many users as possible. But I'm not as convinced, that providing a development environment based on conda is worth the effort. This may be my bias from developing on a Linux system though. And you raise a good point about conda facilitating compilers. I don't really follow how you other arguments are relevant though, since I'm not proposing to drop scikit-image from conda-forge.

My proposal, can we merge this to get the misleading instructions out of the way? I'd be very happy to review a follow-up PR on adding an `environement.yml` and instructions for a conda-forge based dev setup. Or, if you can make an alternative PR soon, merging this might not even be necessary. :)

**Comment by lagru** [comment]
More thoughts:

Maybe we can make this `environment.yml` a lot more restrictive in terms of minimal versions to make it easier to maintain?

I'd like some automation that tells us if the `environment.yml` breaks. Otherwise we will probably just forget this again and -- in the worst case -- have conflicting instructions for dependencies. Do you have ideas how to avoid this case?

File: INSTALL.rst
Comment (lagru) [reply INSTALL.rst:182]: What's the Windows side of this? I would have thought that Linux isn't much of an issue since most of the "help me compile this" issues in my memory are from Windows users.

**Comment by hmaarrfk** [comment]
I don't really want to get into the whole conda vs pip thing. But I do want to highlight one main advantage for linux users + conda-forge over linux users + pip. I will just show an example of ffmpeg, which is often the backend to many frontends that "decode" videos.

On pypi, the one used by imageio-ffmpeg doesn't include support for the nvidia hardware encoders.

<details>

```
./ffmpeg-linux64-v4.2.2 -codecs | grep nv
ffmpeg version 4.2.2-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
 D.V.L. dsicinvideo          Delphine Software International CIN video
 D.V.L. idcin                id Quake II CIN video (decoders: idcinvideo )
 D.VIL. wnv1                 Winnov WNV1
 D.A.L. twinvq               VQF TwinVQ
```
</details>

On conda-forge, it is available. This was due to a multi-year effort of mine, a non-maintainer of upstream's imageio-ffmpeg project and many others at conda-forge to continuously work to expose more and more hardware acceleration from various libraries that enable this.
https://github.com/conda-forge/ffmpeg-feedstock/pull/147

<details>

```
ffmpeg version 6.1.1 Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12.3.0 (conda-forge gcc 12.3.0-11)
  configuration: --prefix=/home/conda/feedstock_root/build_artifacts/ffmpeg_1718893563683/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac --cc=/home/conda/feedstock_root/build_artifacts/ffmpeg_1718893563683/_build_env/bin/x86_64-conda-linux-gnu-cc --cxx=/home/conda/feedstock_root/build_artifacts/ffmpeg_1718893563683/_build_env/bin/x86_64-conda-linux-gnu-c++ --nm=/home/conda/feedstock_root/build_artifacts/ffmpeg_1718893563683/_build_env/bin/x86_64-conda-linux-gnu-nm --ar=/home/conda/feedstock_root/build_artifacts/ffmpeg_1718893563683/_build_env/bin/x86_64-conda-linux-gnu-ar --disable-doc --disable-openssl --enable-demuxer=dash --enable-hardcoded-tables --enable-libfreetype --enable-libharfbuzz --enable-libfontconfig --enable-libopenh264 --enable-libdav1d --enable-gnutls --enable-libmp3lame --enable-libvpx --enable-libass --enable-pthreads --enable-vaapi --enable-libopenvino --disable-gpl --enable-libaom --enable-libsvtav1 --enable-libxml2 --enable-pic --enable-shared --disable-static --enable-version3 --enable-zlib --enable-libopus --pkg-config=/home/conda/feedstock_root/build_artifacts/ffmpeg_1718893563683/_build_env/bin/pkg-config
  libavutil      58. 29.100 / 58. 29.100
  libavcodec     60. 31.102 / 60. 31.102
  libavformat    60. 16.100 / 60. 16.100
  libavdevice    60.  3.100 / 60.  3.100
  libavfilter     9. 12.100 /  9. 12.100
  libswscale      7.  5.100 /  7.  5.100
  libswresample   4. 12.100 /  4. 12.100
 DEV.L. av1                  Alliance for Open Media AV1 (decoders: libdav1d libaom-av1 av1 av1_cuvid) (encoders: libaom-av1 libsvtav1 av1_nvenc av1_vaapi)
 D.V.L. dsicinvideo          Delphine Software International CIN video
 DEV.LS h264                 H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 (decoders: h264 libopenh264 h264_cuvid) (encoders: libopenh264 h264_nvenc h264_vaapi)
 DEV.L. hevc                 H.265 / HEVC (High Efficiency Video Coding) (decoders: hevc hevc_cuvid) (encoders: hevc_nvenc hevc_vaapi)
 D.V.L. idcin                id Quake II CIN video (decoders: idcinvideo)
 D.VIL. wnv1                 Winnov WNV1
 D.AIL. twinvq               VQF TwinVQ
```
</details>

For many others, conda-forge has been a pioneer in supporting osx-arm64 with builds from scikit-image 0.17.2 https://anaconda.org/conda-forge/scikit-image/files?page=11 whereas it was only made available on pypi to osx-arm64 users since 0.19.1


It seems that one of the problems was the change of "flavor" in lazy_loader's spelling, from `lazy_loader` to `lazy-loader` in https://github.com/scikit-image/scikit-image/pull/7373/files


Please see the suggestion in https://github.com/scikit-image/scikit-image/pull/7483 for the suggested 2 line fix.

Honestly, you got an OK from Jarrod on this PR, so you should feel empowered to press merge on either PR. 2 core devs agreeing on a change should be more than enough to make it go through. There is no need to necessarily wait for so much consensus.

**Comment by jni** [comment]
I'm 👎 on this as well. I'm a heavy conda user for many reasons, and I'm just not going to go the pip/venv route, so this change means I'm totally left in the lurch figuring out how to set up my dev environment. Which is fine, I can probably handle it 😂, but probably a symptom of something worse.

Re [synchronising dependency files](https://github.com/scikit-image/scikit-image/pull/7480#discussion_r1702075242), I'd like to request @jaimergp's help for suggestions, since he's done some cool work in that domain for napari. (There is some manual work involved, but we do get warnings.)

Ultimately, though, I see this discussion happening over and over and over, and it's not going to disappear because the tooling is evolving faster than maintainers can update their docs (see micromamba, uv, pixi...). My suggestion therefore would be to upstream this work to the [Scientific Python](https://scientific-python.org/specs/) org, and then we can just say, "if you're using pip+venv, follow these instructions to set up your environment based on this dependency file, if you're using conda environments, follow these other instructions based on this other automatically updated dependency file" (both linking out to SPECs), "then follow these instructions to make sure you have a C++ compiler, then type `spin build --inplace`" (or whatever the invocation is, I don't know these days 😂)

**Comment by lagru** [comment]
> Honestly, you got an OK from Jarrod on this PR, so you should feel empowered to press merge on either PR. 2 core devs agreeing on a change should be more than enough to make it go through. There is no need to necessarily wait for so much consensus.

I somewhat agree, but I think this isn't an issue I need to be pushy about. And giving you the chance to elaborate and push back was definitely worth it IMO. :D

**Comment by lagru** [comment]
And making this PR was useful because it got me feedback from you two which I might not have gotten with an easier to overlook comment on https://github.com/scikit-image/scikit-image/issues/7478. :stuck_out_tongue_winking_eye:

**Comment by lagru** [comment]
I'm closing this but https://github.com/scikit-image/scikit-image/issues/7478 should definitely be addressed. Possibly upstream by https://github.com/conda-forge/lazy_loader-feedstock/pull/11.

**Pull request closed without merging**
